docs/sdk-configuration.md

```diff
@@ -0,0 +1,492 @@
+serverless runtime
+
+### Configuring the SDK in a framework
+
+It is extremely common for Java apps to be written using a dependency injection framework like
+Spring, Guice, Dagger, HK, and many more. They will all follow a very similar pattern though.
+
+```java
+@Component
+public class OpenTelemetryModule {
+  
+  @Bean
+  public OpenTelemetry openTelemetry(MonitoringConfig config) {
+    OpenTelemetrySdk.builder()
+      .setResource(Resource.getDefault().merge(CoolResource.getDefault()))
+      .setClock(AtomicClock.create())
+      .setPropagators(B3Propagator.getInstance())
+      .addSpanExporter(OtlpGrpcSpanExporter.builder()
+        .setEndpoint(config.getOtlpExporter().getEndpoint())
+        .setTimeout(config.getOtlpExporter().getTimeout())
+        .setBatchQueueSize(config.getOtlpExporter().getQueueSize())
+        .build())
+      .addSpanExporter(ZipkinExporter.builder()
+        .setEndpoint(config.getZipkinExporter().getEndpoint())
+        .disableBatching()
+        .build())
+      .addSpanExporter(FullyAsyncZipkinExporter.create())
+      .addSpanProcessor(CustomAttributeAddingProcessor.create())
+      .addSpanProcessor(CustomEventAddingProcessor.create())
+      .setTraceSampler(config.getSamplingRate() != 0 
+          ? Sampler.rateLimiting(config.getSamplingRate()) : Sampler.getDefault())
+      .setTraceLimits(TraceLimits.builder().setMaxAttributes(config.getMaxSpanAttributes()).build())
+      .setIdGenerator(TimestampedIdGenerator.create())
+      .setMeterStuff(InProgressMetricsStuff.create())
+      .build();
+  }
+  
+  @Bean
+  public TracerProvider tracerProvider(OpenTelemetry openTelemetry) {
+    return openTelemetry.getTracerProvider();
+  }
+  
+  @Bean
+  public MeterProvider tracerProvider(OpenTelemetry openTelemetry) {
+    return openTelemetry.getMeterProvider();
+  }
+  
+  @Bean
+  public AuthServiceStub authService(OpenTelemetry openTelemetry, AuthConfig config) {
+    return AuthServiceGrpc.newBlockingStub(ManagedChannelBuilder.forEndpoint(config.getEndpoint()))
+      .withInterceptor(TracingClientInterceptor.create(openTelemetry));
+  }
+  
+  @Bean
+  public ServletFilter servletFilter(OpenTelemetry openTelemetry) {
+    return TracingServletFilter.create(openTelemetry);
+  }
+}
+
+// Use some instrumented client
+@Component
+public class MyAuthInterceptor {
+
+  private final AuthServiceStub authService;
+  
+  @Inject
+  public MyAuthInterceptor(AuthServiceStub authService) {
+    this.authService = authService;
+  }
+  
+  public void doAuth() {
+    if (!authService.getToken("credential").isAuthenticated()) {
+      throw new HackerException();
+    }
+  }
+}
+
+// Use tracer directly, not so common
+@Component
+public class MyService {
+
+  private final Tracer tracer;
+  private final Meter meter;
+  
+  @Inject
+  public MyService(TracerProvider tracerProvider, MeterProvider meterProvider) {
```

Comment:
I would encourage passing Otel instance always. So if we add more functionality to the otel instance we don't break the API.